lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1046,8 +1046,13 @@ private function executeUpdates($class)
             }
 
             if ($preUpdateInvoke != ListenersInvoker::INVOKE_NONE) {
-                $this->listenersInvoker->invoke($class, Events::preUpdate, $entity, new PreUpdateEventArgs($entity, $this->em, $this->getEntityChangeSet($entity)), $preUpdateInvoke);
-
+                if (!array_key_exists($oid, $this->entityChangeSets) || is_null($this->entityChangeSets[$oid])) {
```

Comment:
1. I agree with the `isset`
2. I need a variable because it is passed by reference, it fails when I pass directly a value (with `array()` or with `[]`)